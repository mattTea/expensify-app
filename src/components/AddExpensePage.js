import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { addExpense } from '../actions/expenses';

export class AddExpensePage extends React.Component { // <- export so it's possible to test the unconnected component
  onSubmit = (expense) => {
    this.props.addExpense(expense);
    this.props.history.push('/'); // <- method available that pushes you to a path in the app
  };
  render() {
    return (
      <div>
        <h1>Add Expense</h1>
        <ExpenseForm
          onSubmit={this.onSubmit}
        />
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  addExpense: (expense) => dispatch(addExpense(expense))
});

export default connect(undefined, mapDispatchToProps)(AddExpensePage);
