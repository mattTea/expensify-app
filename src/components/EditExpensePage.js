import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { editExpense, removeExpense } from '../actions/expenses';

const EditExpensePage = (props) => {
  return (
    <div>
      <ExpenseForm
        expense={props.expense}
        onSubmit={(expense) => {
          props.dispatch(editExpense(props.expense.id, expense));
          props.history.push('/');
        }}
      />
      <button onClick={() => {
        props.dispatch(removeExpense({ id: props.expense.id }));
        console.log(props.expense.id);
        props.history.push('/');
      }} >Remove</button>
    </div>
  );
}

const mapStateToProps = (state, props) => {
  return { // <- add expense prop onto passed in props and set it to equal the current expense state using array.find() method
    expense: state.expenses.find((expense) => expense.id === props.match.params.id)
  };
};

export default connect(mapStateToProps)(EditExpensePage);