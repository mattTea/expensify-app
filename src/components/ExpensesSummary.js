import React from 'react';
import numeral from 'numeral';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = (props) => (
  <div>
    {console.log(props.expenseCount)}
    {
      !props.expenseCount || props.expenseCount.length === 0 ? (
        <p>No expenses to view</p>
      ) : (
        <p>{`Viewing ${props.expenseCount.length} expenses totalling ${numeral(selectExpensesTotal(props.expenseCount) / 100).format('$0,0.00')}`}</p>
      )
    }
  </div>
);

const mapStateToProps = (state) => {
  return {
    expenseCount: selectExpenses(state.expenses, state.filters)
    // expensesTotal: selectExpensesTotal(state.expenses)
  }
};

export default connect(mapStateToProps)(ExpensesSummary);