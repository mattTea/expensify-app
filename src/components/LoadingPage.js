import React from 'react';

const LoadingPage = () => (
  <div className="loader">
    <div className="loader-spinner"></div>
  </div>
)

export default LoadingPage;