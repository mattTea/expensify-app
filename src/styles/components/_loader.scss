.loader {
  align-items: center;
  display: flex;
  height: 100vh;
  justify-content: center;
  width: 100vw;
}

.loader-spinner {
  animation: zRotate 2s infinite linear;
  border: solid 3px $off-white;
  border-bottom-color: $blue;
  border-radius: 50%;
  height: 6rem;
  width: 6rem;
}

@keyframes zRotate {
  to {
    transform: rotateZ(1turn);
  }
}