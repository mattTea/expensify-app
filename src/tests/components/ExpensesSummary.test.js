import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should render ExpensesSummary correctly with 1 expense', () => {
  const wrapper = shallow(<ExpensesSummary />);
  expect(wrapper).toMatchSnapshot();
});