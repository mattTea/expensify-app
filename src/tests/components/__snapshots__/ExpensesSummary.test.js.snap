// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary correctly with 1 expense 1`] = `
<div>
  <p>
    No expenses to view
  </p>
</div>
`;
